src/Microsoft.Health.Dicom.Api/Features/Converter/EnumNameJsonConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Api.Features.Converter
+{
+    /// <summary>
+    /// Enum JsonConverter to provide better error message to customer.
+    /// </summary>
+    /// <typeparam name="T">Enum type.</typeparam>
+    public sealed class EnumNameJsonConverter<T> : JsonConverter<T>
+        where T : Enum
+    {
+        public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            EnsureArg.IsNotNull(typeToConvert, nameof(typeToConvert));
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                string content = reader.GetString();
+                if (Enum.TryParse(typeToConvert, content, true, out object result))
```

Comment:
After playing around, it appears that the `Enum` type constraint doesn't imply `struct`. So if you simply change your type constraint at the top to `where T : struct, Enum` then you should be able to use `Enum.TryParse<T>(content, true, out T result)`